### PR TITLE
Stop app init if too many sentences

### PR DESCRIPTION
**Pull request by sjonany**
Don't initialize app if
1. readability doesn't parse the page (e.g. image) or
2. too many sentences (which can cause browser to crash for crappy computers)

### CODE REVIEW

File: extension/lib/doc.js [hunk 1]
```diff
@@ -7,6 +7,9 @@ if (window[namespace] === true) {
     window[namespace] = true;
 }
 
+// If the number of sentences in the doc exceeds this, mark the document as unreadable.
+const MAX_NUM_SENTENCE = 1000;
+
```
Comment (benfrancis314) [inline extension/lib/doc.js:12]: It'd be interesting to see how long an article would be for this to be the case; for instance, what about a long Wikipedia article? An online book? Also related to speed function, i.e. the function time_to_load(#_of_sentences)

File: extension/lib/doc.js [hunk 2]
```diff
@@ -103,6 +107,13 @@ class Doc {
                 let end = start + sentenceBoundary.offset;
                 this.sentences.push(new SentencePointer(container_id, start, end));
             }
+            if (this.sentences.length > MAX_NUM_SENTENCE) {
+                debug("Too many sentences found = " + this.sentences.length);
```
Comment (benfrancis314) [inline extension/lib/doc.js]: Should we possibly alert this to them, so they're not confused if this happens?

**Review by benfrancis314 — APPROVED** [review]
LGTM

File: extension/lib/doc.js (hunk 2 above)
Comment (sjonany) [reply extension/lib/doc.js]: Yeah good point. Hmm.. for now the quick options are either:
1. alert popup
2. console log

I say alert cos altho unsightly, at least they won't think that the app is busted.
Also, not that annoying cos we tell them just once per big page and they'll learn.

**Pull request merged**